pennylane/devices/default_tensor.py

```diff
@@ -388,7 +388,9 @@ def _apply_operation(self, op: qml.operation.Operator) -> None:
             op (Operator): The operation to apply.
         """
 
-        self._circuitMPS.apply_gate(op.matrix().astype(self._dtype), *op.wires, **self._gate_opts)
+        self._circuitMPS.apply_gate(
+            qml.matrix(op).astype(self._dtype), *op.wires, **self._gate_opts
```

Comment:
I hope the answer is "no", but could this lead to performance deprecation? :grimacing:

Reply:
That's a good point @dwierichs 

> could this lead to performance deprecation?

I most certainly don't think so. I did a few quick tests and it seems that for a `Hamiltonian` instance there's no difference. This is not surprising since the `qml.matrix` method [just calls the corresponding matrix() method](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/ops/functions/matrix.py#L220) in case the argument is an operator

Reply:
Nice, thanks!